src/features/surveys/hooks/useSurveySubmission.ts

```diff
@@ -18,17 +18,20 @@ type UseSurveySubmissionReturn = IFuture<ZetkinSurveySubmission> & {
 
 export default function useSurveySubmission(
   orgId: number,
+  surveyId: number,
   submissionId: number
 ): UseSurveySubmissionReturn {
   const apiClient = useApiClient();
   const dispatch = useAppDispatch();
 
   const submissionItem = useAppSelector((state) =>
-    state.surveys.submissionList.items.find((item) => item.id == submissionId)
+    state.surveys.submissionsBySurveyId[surveyId].items.find(
+      (item) => item.id == submissionId
+    )
```

Comment:
Instead of finding just here, we could loop over all of the `submissionsBySurveyId` entries and search each of them until the correct submission is found (or not found, which means it has not been loaded and `submissionItem` should become falsy, so that's fine).